internal/pkg/template/template.go

```diff
@@ -63,6 +94,34 @@ func (t *Template) Parse(path string, data interface{}, options ...ParseOption)
 	return &Content{buf}, nil
 }
 
+// UploadEnvironmentCustomResources uploads the environment custom resource scripts.
+func (t *Template) UploadEnvironmentCustomResources(upload func(string, ...CustomResource) (string, error)) ([]CustomResource, error) {
```

Comment:
In the future, I expect us to also have `UploadEnvironmentTemplate` and maybe `UploadWorkloadTemplate`. I think it's fine if we have separate public methods for each one of these, but we should be able to call into a common private method.

Right now `uploadCustomResources` is specific to custom resources by taking `...CustomResource` as input and returning `[]CustomResource` as output.

Instead, can we think of a way of maybe leveraging interfaces or basic types to make this more common?

```go
// Just brainstorming here:

type Uploadable interface {
   Key() string
   Body() []byte
}
type UploadFn func(Uploadable) (url string, err error)

func (t *Template) uploadFiles(files ...uploadable, fn UploadFn) (urls []string, err error)
```